### ISSUE TITLE
eclipse update of maven project fails for sonar-cxx-plugin

### ISSUE THREAD

**Issue by Bertk** [SonarOpenCommunity/sonar-cxx#851]
Eclipse fails to update the project definition from pom.xml
```
!ENTRY org.eclipse.m2e.core 4 0 2016-05-01 16:52:54.074
!MESSAGE Could not update project sonar-cxx-plugin configuration
!STACK 0
java.lang.NullPointerException
        at org.eclipse.m2e.jdt.internal.AbstractJavaProjectConfigurator.isContained(AbstractJavaProjectConfigurator.java:470)
        at org.eclipse.m2e.jdt.internal.AbstractJavaProjectConfigurator.isNonOverlappingResourceDescriptor(AbstractJavaProjectConfigurator.java:461)
        at org.eclipse.m2e.jdt.internal.AbstractJavaProjectConfigurator.addResourceDirs(AbstractJavaProjectConfigurator.java:422)
        at org.eclipse.m2e.jdt.internal.AbstractJavaProjectConfigurator.addProjectSourceFolders(AbstractJavaProjectConfigurator.java:299)
        at org.eclipse.m2e.jdt.internal.AbstractJavaProjectConfigurator.configure(AbstractJavaProjectConfigurator.java:122)
        at org.eclipse.m2e.core.project.configurator.AbstractLifecycleMapping.configure(AbstractLifecycleMapping.java:120)
        at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$3.call(ProjectConfigurationManager.java:501)
        at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$3.call(ProjectConfigurationManager.java:1)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:176)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
        at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.updateProjectConfiguration(ProjectConfigurationManager.java:494)
        at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.configureNewMavenProjects(ProjectConfigurationManager.java:281)
        at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$1.call(ProjectConfigurationManager.java:168)
        at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager$1.call(ProjectConfigurationManager.java:1)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:176)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
        at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:99)
        at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:1355)
        at org.eclipse.m2e.core.internal.project.ProjectConfigurationManager.importProjects(ProjectConfigurationManager.java:136)
        at org.eclipse.m2e.core.ui.internal.wizards.ImportMavenProjectsJob$1.doCreateMavenProjects(ImportMavenProjectsJob.java:68)
        at org.eclipse.m2e.core.ui.internal.wizards.AbstractCreateMavenProjectsOperation.run(AbstractCreateMavenProjectsOperation.java:62)
        at org.eclipse.m2e.core.ui.internal.wizards.ImportMavenProjectsJob.runInWorkspace(ImportMavenProjectsJob.java:77)
        at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:39)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
```
The pom.xml shows a resource using a location outside of the project folder which causes the issue. After removing the resource from the pom.xml the update works again.
```xml
  
      <resource>
        <targetPath>static</targetPath>
        <directory>../cxx-lint/target</directory>
        <includes>
          <include>cxx-lint-${project.version}.jar</include>
        </includes>
      </resource>

```

**Comment by guwirth** [comment]
@jmecosta maybe you can fix it?

**Comment by jmecosta** [comment]
i dont know, but seems to work with netbeans and maven command line. perhaps some eclipse bug? seems a standard maven feature and we dont have issues with it. is there more people having same issue with other ides?

**Comment by guwirth** [comment]
I'm using NetBeans and Maven: it's working with both.

**Comment by Bertk** [comment]
I modified the pom.xml and this version works fine. Please check.
_I did not remove the SQ version for the sonar-plugin-api or sonar-check-api but this should be defined in the parent pom.xml_
```XML
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.codehaus.sonar-plugins.cxx</groupId>
    <artifactId>cxx</artifactId>
    <version>1.0-SNAPSHOT</version>
  </parent>
  <artifactId>cxx-lint</artifactId>
  <name>Cxx :: Lint</name>
  <packaging>jar</packaging>
  <dependencies>
    <dependency>
      <groupId>${project.groupId}</groupId>
      <artifactId>cxx-checks</artifactId>
      <version>${project.version}</version>
    </dependency>
    <dependency>
      <groupId>com.google.code.gson</groupId>
      <artifactId>gson</artifactId>
      <version>2.4</version>
    </dependency>
    <dependency>
      <groupId>org.codehaus.sonar</groupId>
      <artifactId>sonar-check-api</artifactId>
      <version>4.5.4</version>
      <type>jar</type>
    </dependency>
    <dependency>
      <groupId>org.codehaus.sonar</groupId>
      <artifactId>sonar-plugin-api</artifactId>
      <version>4.5.4</version>
      <type>jar</type>
    </dependency>
    <dependency>
      <groupId>commons-cli</groupId>
      <artifactId>commons-cli</artifactId>
      <version>1.3.1</version>
      <type>jar</type>
    </dependency>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.12</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.hamcrest</groupId>
      <artifactId>hamcrest-core</artifactId>
      <version>1.3</version>
      <scope>test</scope>
    </dependency>
  </dependencies>

  <build>
    <plugins>
      <plugin>
        <artifactId>maven-assembly-plugin</artifactId>
        <configuration>
          <archive>
            <manifest>
              <addClasspath>true</addClasspath>
              <mainClass>org.codehaus.sonarplugins.cxx.cxxlint.CxxLint</mainClass>
              <classpathPrefix>lib/</classpathPrefix>
            </manifest>
          </archive>
          <descriptorRefs>
            <descriptorRef>jar-with-dependencies</descriptorRef>
          </descriptorRefs>
          <finalName>cxx-lint-${project.version}</finalName>
          <appendAssemblyId>false</appendAssemblyId>
        </configuration>
        <executions>
          <execution>
            <id>make-assembly</id>
            <!-- this is used for inheritance merges -->
            <phase>package</phase>
            <!-- bind to the packaging phase -->
            <goals>
              <goal>single</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>

</project>
      
</project>
```

**Comment by jmecosta** [comment]
@Bertk i will create a pr to and test in netbeans, dont see any reason not to merge this change

**Comment by jmecosta** [comment]
@Bertk are you sure the pom you posted is correct? as far as i see there are no differences to current master.

![image](https://cloud.githubusercontent.com/assets/1041352/15098416/e93c71bc-1545-11e6-883e-34eec8e7bc06.png)

**Comment by Bertk** [comment]
@jmecosta : OK let us close this issue. I cannot find the strange path in the pom.xml.
```XML
      <resource>
        <targetPath>static</targetPath>
        <directory>../cxx-lint/target</directory>
        <includes>
          <include>cxx-lint-${project.version}.jar</include>
        </includes>
      </resource>
```

**Comment by jmecosta** [comment]
But that's the pom for the cxx-plugin perhaps you need to share that one

**Comment by Bertk** [comment]
@jmecosta Thanks for the hint. I updated the comment with the pom.xml details and now it should be clear :sunglasses:

**Comment by jmecosta** [comment]
fixed by #863
